src/MergeContext.js

```diff
@@ -75,6 +77,69 @@ class PrProblem extends Error {
     }
 }
 
+// returns a string with Unicode characters replaced with their
+// hexadecimal Unicode code points
+function EscapeUnsafeCharacters(str) {
+    return str.replace(new RegExp(Util.PrMessageProhibitedCharacters, 'gu'), (uchar) => {
+        // Pad with leading zeros to ensure 4 digits (e.g., \u00E9)
+        const encoded = uchar.charCodeAt(0).toString(16).toUpperCase().padStart(4, '0');
+        return '\\u' + encoded;
+    });
+}
+
+// A PrProblem caused by PR title or description content.
+class PrDescriptionProblem extends PrProblem
+{
+    constructor(parsingContext, errorDescription, problematicInput, ...params) {
+        assert(parsingContext);
+        assert(errorDescription);
+        super(errorDescription, ...params);
+        this._parsingContext = parsingContext;
+        this._errorDescription = errorDescription;
+        this._problematicInput = problematicInput; // may be null
+    }
+
+    // creates a Markdown text suitable for being a second GitHubUtil::createComment() parameter
+    toGitHubComment() {
+        let errorMessage = `Cannot create a git commit message from PR title and description.\n\n`;
+        errorMessage += `Invalid PR description ${this._parsingContext}: `;
+
+        if (this._errorDescription === ErrorDescriptionInvalidCharacter) {
+            const match = Util.PrMessageProhibitedCharacters.exec(this._problematicInput);
+            assert(match);
+            const escaped = EscapeUnsafeCharacters(this._problematicInput);
+            // the encoded length is 6: e.g., \u00E9
+            const badCharEncoded = escaped.substring(match.index, match.index + 6);
+            errorMessage += `\`Invalid ${this._parsingContext} character (Unicode ${badCharEncoded}) at position ${match.index}\`:\n`;
+        } else {
+            errorMessage += `\`${this._errorDescription}\`:\n`;
+        }
+
+        if (this._problematicInput) {
+            const escaped = EscapeUnsafeCharacters(this._problematicInput);
+            // indent with 4 spaces to ask GitHub to render user input without Markdown formatting
+            errorMessage += `\n    ${escaped}\n`;
```

Comment:
To give humans a better understanding of what Anubis was looking at when generating errors, I would remove trailing `:` from errorMessage lines above and then explicitly name this blob as "parser input". Something like this may work:

```suggestion
            errorMessage += `\nParser input:\n    ${escaped}\n`;
```

or even

```suggestion
            errorMessage += `\nProblematic parser input:\n    ${escaped}\n`;
```